### ISSUE TITLE
Issue with installing and upgrading some plugins

### ISSUE THREAD

**Issue by AvianY** [vim-volt/volt#207]
I used the command $volt get w0rp/ale so I could install it, but I get back the message:

[INFO] Full building /home/aviany/.vim/pack/volt/opt directory ...
[INFO] Installing vimrc and gvimrc ...
! github.com/w0rp/ale > install failed
  * failed to get HEAD commit hash: reference not found
[ERROR] failed to install some plugins

similarly, when I try to update existing plugins using $volt -l -u I get an error message:
[INFO] Full building /home/aviany/.vim/pack/volt/opt directory ...
[INFO] Installing vimrc and gvimrc ...
! github.com/davidhalter/jedi-vim > upgrade failed
  * failed to upgrade plugin: gitconfig 'branch.master.remote' is not found
[ERROR] failed to install some plugins

other plugins seem to have no issue, does anyone know why is this happening?

**Comment by AvianY** [comment]
I managed to fix it.... apparently both plugins required installing them with full https://www.github.com/user/plugin name instead of just /user/plugin name. This way, they both install and update just fine....

**Comment by tyru** [comment]
> apparently both plugins required installing them with full https://www.github.com/user/plugin name instead of just /user/plugin name

Hmm, that is strange.
Basically, full name or abbreviation name should not affect installing/upgrading behavior.
(FYI, abbreviation of `https://www.github.com/user/plugin` is `user/plugin`, not `/user/plugin`)

> failed to get HEAD commit hash: reference not found

Does this problem still occur in your environment? (by uninstalling w0rp/ale with `volt rm -r -p w0rp/ale`, and install again with `volt get w0rp/ale`)
In my environment, I can't reproduce it.

> failed to upgrade plugin: gitconfig 'branch.master.remote' is not found

I guess it's because installation was failed.
The repository might be invalid state, but I don't know the reason yet...

**Comment by AvianY** [comment]
Thanks for your response. I have run the commands as you said, and now it works. I managed to get all the plugins to be in the 'github.com' folder

**Comment by AvianY** [comment]
Actually, let me correct myself:
There is something a bit strange going on... I removed those two plugins (w0rp/ale and davidhalter/jedi-vim) again using your command and also went into the folders to manually delete the confs and repos.
Then I fresh installed them with the other command and they install fine
BUT,
when I try to run "volt get -l -u" to update all plugins, these two give a warning message:
[WARN] failed to clone, try to execute "git clone --recursive https://www.github.com/davidhalter/jedi-vim /home/aviany/.vim/volt/repos/www.github.com/davidhalter/jedi-vim" instead...: empty git-upload-pack given
[WARN] failed to clone, try to execute "git clone --recursive https://www.github.com/w0rp/ale /home/aviany/.vim/volt/repos/www.github.com/w0rp/ale" instead...: empty git-upload-pack given

and on the end of text, they give:
+ www.github.com/davidhalter/jedi-vim > installed
+ www.github.com/w0rp/ale > installed

which again installs them in a separate "www.github.com" directory and also duplicates the same plugin.
I don't know why is it failing to clone only during an update...

**Comment by tyru** [comment]
Could you give me `~/volt/lock.json` file?

And I was wrong, w0rp/ale is abbreviation to github.com/w0rp/ale, not www.github.com/w0rp/ale .

https://github.com/vim-volt/volt/blob/8f2139fa82e245a20834c75405d4449084d79961/pathutil/pathutil.go#L37

please use abbreviated notation or github.com/w0rp/ale .

> also went into the folders to manually delete the confs and repos.

`volt rm -r -p` deletes also confs(-p) and repos(-r).
If confs and repos are not deleted, it's a bug.

> which again installs them in a separate "www.github.com" directory and also duplicates the same plugin.

Do you mean that plugin directories are duplicate in "www.github.com" and "github.com" directories?

**Comment by AvianY** [comment]
[lock.zip](https://github.com/vim-volt/volt/files/1840633/lock.zip)

Here is my $VOLTPATH/lock.json zipped.

I tried again the command "volt rm -r -p" and it works. I think what happened was, that I deleted w0rp/ale plugin in the github.com folder, but was surprised to still see them still in the www.github.com folder, so I deleted them manually, even though I should separately delete them with another command, so there is no issue...

When I said duplicate, I meant, that there is the exact same plugin in the github.com and www.github.com directories. I don't know if that's a problem for volt or not, but it's confusing to me...
